File: backend/ws-service/src/main/java/com/example/wsservice/chat/dto/ChatRoomDto.java
package com.example.wsservice.chat.dto;

import com.example.wsservice.member.dto.MemberDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChatRoomDto implements Serializable {
    private String roomId;
    private String roomName;
    private String roomImage;
    private String lastChatMsg;
    private String lastChatId;
    private String lastChatTime;
    private List<MemberDto> members = new ArrayList<>();

    public void updateLastChat(String chatId, ChatDto chatDto) {
        setLastChatId(chatId);
        setLastChatMsg((chatDto.getChatType() == ChatType.TEXT.getChatType()) ?
                chatDto.getMessage() : ChatType.fromChatType(chatDto.getChatType()).getChatTypeStr());
        setLastChatTime(chatDto.getChatTime());
    }

    public boolean checkChatRoomMember(String userId) {
        List<MemberDto> chatRoomMembers = members.stream()
                .filter(memberDto -> memberDto.getUserId().equals(userId))
                .collect(Collectors.toList());

        return chatRoomMembers.size() == 0;
    }
}

File: backend/ws-service/src/main/kotlin/com/example/wsservice/chat/dto/ChatRoomDto.kt
package com.example.wsservice.chat.dto

import com.example.wsservice.member.dto.MemberDto
import java.io.Serializable

data class ChatRoomDto(
    var roomId: String? = null,
    var roomName: String? = null,
    var roomImage: String? = null,
    var lastChatMsg: String? = null,
    var lastChatId: String? = null,
    var lastChatTime: String? = null,
    var members: List<MemberDto>? = ArrayList(),
) : Serializable {

    fun updateLastChat(chatId: String, chatDto: ChatDto) {
        lastChatId = chatId
        lastChatMsg = if (chatDto.chatType == ChatType.TEXT.chatType) chatDto.message else ChatType.fromChatType(chatDto.chatType)?.chatTypeStr
        lastChatTime = chatDto.chatTime
    }

    /** true 면 방 멤버가 아니다(옛 자바 이름 그대로). */
    fun checkChatRoomMember(userId: String?): Boolean = members.orEmpty().none { it.userId == userId }
}
